xdem/dem.py

```diff
@@ -105,151 +128,103 @@ def copy(self, new_array: NDArrayf | None = None) -> DEM:
-                self.vref_grid = vref_grid
-            else:
-                if os.path.exists(os.path.join(pyproj.datadir.get_data_dir(), vref_grid)):
-                    self.vref = "Unknown vertical reference name from: " + vref_grid
-                    self.vref_grid = vref_grid
-                else:
-                    raise ValueError(
-                        "Grid not found in " + str(pyproj.datadir.get_data_dir()) + ": check if proj-data is "
-                        "installed via conda-forge, the pyproj.datadir, and that you are using a grid available at "
-                        "https://github.com/OSGeo/PROJ-data"
-                    )
-
-        # Otherwise, use name provided
-        elif isinstance(vref_name, str):
-            if vref_name == "WGS84":
-                self.vref_grid = None
-                self.vref = "WGS84"  # WGS84 ellipsoid
-            elif vref_name == "EGM08":
-                self.vref_grid = "us_nga_egm08_25.tif"  # EGM2008 at 2.5 minute resolution
-                self.vref = "EGM08"
-            elif vref_name == "EGM96":
-                self.vref_grid = "us_nga_egm96_15.tif"  # EGM1996 at 15 minute resolution
-                self.vref = "EGM96"
-            else:
-                raise ValueError(
-                    'Vertical reference name must be either "WGS84", "EGM96" or "EGM08". Otherwise, provide'
-                    " a geoid grid from PROJ DATA: https://github.com/OSGeo/PROJ-data"
-                )
+    @property
+    def ccrs(self) -> CompoundCRS | CRS | None:
+        """Compound horizontal and vertical coordinate reference system of the DEM."""
 
-        # Else, return an error
+        if self.vcrs is not None:
+            ccrs = _build_ccrs_from_crs_and_vcrs(crs=self.crs, vcrs=self.vcrs)
+            return ccrs
         else:
-            raise ValueError("Vertical reference name or vertical grid must be a string")
-
-    def to_vref(self, vref_name: str = "EGM96", vref_grid: str | None = None) -> None:
+            return None
 
+    def to_vcrs(
+        self,
+        dst_vcrs: Literal["Ellipsoid", "EGM08", "EGM96"] | str | pathlib.Path | VerticalCRS | int,
+        src_vcrs: Literal["Ellipsoid", "EGM08", "EGM96"] | str | pathlib.Path | VerticalCRS | int | None = None,
+    ) -> None:
         """
-        Convert between vertical references: ellipsoidal heights or geoid grids.
+        Convert the DEM to another vertical coordinate reference system.
 
-        :param vref_name: Vertical reference name
-        :param vref_grid: Vertical reference grid (any grid file in https://github.com/OSGeo/PROJ-data)
+        :param dst_vcrs: Destination vertical CRS. Either as a name ("WGS84", "EGM08", "EGM96"),
+            a EPSG code or pyproj.crs.VerticalCRS, or a path to a PROJ grid file (https://github.com/OSGeo/PROJ-data)
+        :param src_vcrs: Force a source vertical CRS (uses metadata by default). Same formats as for `dst_vcrs`.
 
         :return:
         """
 
```

Comment:
Might be good here to test if `(src_vcrs or self.vcrs) == dst_vcrs` to avoid running this unnecessarily. Imagine if you want to compare two DEMs and want to make sure that they have a vcrs beforehand. Then this might be called even though it may already be set properly.

Reply:
I guess technically this check would be better further down when `src_vcrs` is a `VerticalCRS` and nothing else.

Reply:
Done, it now also raises a `UserWarning` to tell the user that the operation did nothing. And I added tests for it! :smile:

Reply:
In [ad871ba](https://github.com/GlacioHack/xdem/pull/350/commits/ad871baef73bac8e714772625c2f013d5aa3d877)